### PR TITLE
Fix empty vector argument array construction

### LINKED ISSUE #205: Indexing of an empty vector
https://github.com/Gecode/gecode/blob/027c57889d66dd26ad8e1a419c2cda22ab0cf305/gecode/kernel/data/array.hpp#L1571
This line will throw an unexpected exception if `aa` is empty.

### PR DESCRIPTION
**Pull request by zayenz**
## Summary
- Fix construction of argument arrays from empty `std::vector` instances by avoiding `&aa[0]` when the vector is empty.
- Add a regression test for empty vector construction of `IntArgs`, `IntVarArgs`, and `BoolVarArgs`.
- Add a changelog entry thanking @intractabilis.

Fixes #205.

## Validation
- `cmake -S . -B /tmp/gecode-empty-vector-args-build -DCMAKE_BUILD_TYPE=Debug -DGECODE_ENABLE_EXAMPLES=OFF -DGECODE_ENABLE_GIST=OFF -DGECODE_ENABLE_QT=OFF`
- `cmake --build /tmp/gecode-empty-vector-args-build --target gecode-test -j 8`
- `/tmp/gecode-empty-vector-args-build/bin/gecode-test -test Array::Vector::Empty`
- `git diff --check`

### CODE REVIEW

File: gecode/kernel/data/array.hpp [hunk 1]
```diff
@@ -1557,7 +1557,8 @@ namespace Gecode {
   ArgArrayBase<T>::ArgArrayBase(const std::vector<T>& aa)
     : n(static_cast<int>(aa.size())),
       capacity(n < onstack_size ? onstack_size : n), a(allocate(n)) {
-    heap.copy<T>(a,&aa[0],n);
+    if (n > 0)
+      heap.copy<T>(a,aa.data(),n);
```
Comment (kalvdans) [inline gecode/kernel/data/array.hpp:1561]: Nit: aa.data() is fine to call on an empty std::vector (might or might not return nullptr). do we need the `if (n > 0)` then?

**Review by kalvdans — APPROVED** [review]

File: gecode/kernel/data/array.hpp (hunk 1 above)
Comment (zayenz) [reply gecode/kernel/data/array.hpp:1561]: Good point, and thanks for noting that. I still decided to keep the redundant if, as it shouldn't cost anything at runtime, and I like that it is explicit and not relying on API promises.

Comment (intractabilis) [reply gecode/kernel/data/array.hpp:1561]: The canonical way is
```cpp
if (!aa.empty())
    heap.copy<T>(a,aa.data(),n);
```
